### PR TITLE
task_manager_using_typescript

### PR DESCRIPTION
**Pull request by Shiveshtrivedi**
Css units - eithier in  px or %.
No console  value  and commented code.
Document formatted.
Sorting functionality implemented.
No duplicate value allowed.

## Summary by CodeRabbit

- **New Features**
  - Introduced enumerations for filtering task statuses and date formats, enhancing task management capabilities.

- **Chores**
  - Set up Husky for Git hooks to automate linting and formatting checks before commits.
  - Introduced a `.gitignore` file to maintain a clean repository by excluding unnecessary files.

### CODE REVIEW

File: task_manager_using_typescript/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "t",
```
Comment (aakash-soni1) [inline task_manager_using_typescript/package.json]: Give proper value in name field for package.json

File: task_manager_using_typescript/src/data_file.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { InterfaceTask } from "./interfaces";
+
+let baseUrl: string = "https://66a0bb4b7053166bcabc8698.mockapi.io/Tasklist";
```
Comment (aakash-soni1) [inline task_manager_using_typescript/src/data_file.ts]: Make this let variable as const

File: task_manager_using_typescript/src/data_file.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+export const fetchData = <T>(
+  url: string = "",
+  option: RequestInit = {}
+): Promise<T> => {
+  const fullUrl = `${baseUrl}${url}`;
+
+  return fetch(fullUrl, option)
```
Comment (aakash-soni1) [inline task_manager_using_typescript/src/data_file.ts:9]: What if the response is ok, with some error will it goes to handle by catch

File: task_manager_using_typescript/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+  let tasksToDisplay: InterfaceTask[];
+
+  switch (filterType) {
+    case "completed":
+      tasksToDisplay = allTasks.filter((task) => task.isCompleted);
+      break;
+    case "incomplete":
+      tasksToDisplay = allTasks.filter((task) => !task.isCompleted);
+      break;
+    case "favorite":
```
Comment (aakash-soni1) [inline task_manager_using_typescript/src/index.ts]: use ENUM for switch cases

File: task_manager_using_typescript/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+
+const sortTasksByName = (ascending: boolean = true): void => {
+  allTasks.sort((a, b) => {
+    const nameA: string = a.name.toLowerCase();
+    const nameB: string = b.name.toLowerCase();
+    const comparison: number = nameA.localeCompare(nameB);
+    return ascending ? comparison : -comparison;
+  });
+};
+type FilterType = "all" | "completed" | "incomplete" | "favorite";
```
Comment (aakash-soni1) [inline task_manager_using_typescript/src/index.ts]: use ENUM here also

File: task_manager_using_typescript/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+  const actionContainer: HTMLDivElement = document.createElement("div");
+  actionContainer.className = "actionContainer";
+
+  const date: HTMLParagraphElement = document.createElement("p");
+  date.className = "date";
+  const now: Date = new Date();
+  const options: Intl.DateTimeFormatOptions = {
+    year: "numeric",
+    month: "long",
+    day: "numeric",
```
Comment (aakash-soni1) [inline task_manager_using_typescript/src/index.ts]: use ENUM for constant values

File: task_manager_using_typescript/src/interfaces.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  name: string;
+  isCompleted: boolean;
+  isFavorite: boolean;
+}
+
+export interface InterfaceAllTasks {
+  name: string;
+  isCompleted: boolean;
+  isFavorite: boolean;
+}
```
Comment (aakash-soni1) [inline task_manager_using_typescript/src/interfaces.ts]: You can extend the `InterfaceAllTasks ` with `InterfaceTask` as some most of the fields are redundant

**Review by aakash-soni1 — CHANGES_REQUESTED** [review]

File: task_manager_using_typescript/src/data_file.ts (hunk 3 above)
Comment (Shiveshtrivedi) [reply task_manager_using_typescript/src/data_file.ts:9]: If data is  encrypted then promise is rejected and it goes to catch block.

**Review by aakash-soni1 — APPROVED** [review]

File: task_manager_using_typescript/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+
+addTaskCloseButton.addEventListener('click', (): void => {
+  addTaskPopupOverlay.style.display = 'none';
+});
+
+enum FilterTypeValue {
+  completed = 'completed',
+  incomplete = 'incomplete',
+  favorite = 'favorite',
+  All = 'all',
```
Comment (aakash-soni1) [inline task_manager_using_typescript/src/index.ts]: Key of enum should be in uppercase

File: task_manager_using_typescript/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,382 @@
...
+};
+
+const isDuplicateTaskName = (name: string): boolean => {
+  return allTasks.some(
+    (task: InterfaceAllTasks) => task.name.toLowerCase() === name.toLowerCase()
+  );
+};
+enum DateFormat {
+  Long = 'long',
+  Short = 'short',
```
Comment (aakash-soni1) [inline task_manager_using_typescript/src/index.ts]: key of enum should be uppercase

File: task_manager_using_typescript/src/interfaces.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+export interface InterfaceAllTasks {
```
Comment (aakash-soni1) [inline task_manager_using_typescript/src/interfaces.ts]: rename `InterfaceAllTasks ` in format `IAllTasks` or `AllTasks`

**Review by aakash-soni1 — CHANGES_REQUESTED** [review]
Create a separate file for types which include interface, enums and other types.

File: .github/workflows/build.yml [hunk 11]
```diff
@@ -0,0 +1,51 @@
+name: Simple Task manger
```
[the remark this answers is not in the document]
Comment (aakash-soni1) [reply .github/workflows/build.yml]: please fix this typo issue

File: task_manager_using_typescript/src/utils/interfaces.js [hunk 12]
```diff
@@ -0,0 +1 @@
+export {};
```
Comment (aakash-soni1) [inline task_manager_using_typescript/src/utils/interfaces.js]: Why this js file is not excluded ?

**Review by aakash-soni1 — CHANGES_REQUESTED** [review]

**Pull request merged**